cpp/src/arrow/dataset/api.h

```diff
@@ -36,4 +36,7 @@
 #ifdef ARROW_PARQUET
 #include "arrow/dataset/file_parquet.h"
 #endif
+#ifdef ARROW_JSON
+#include "arrow/dataset/file_json.h"
+#endif
```

Comment:
These lines are already included above, no need to add them again.

Reply:
Ah, must have crept in when I rebased, good catch!